tests/e2e-pw/tests/product-collection/product-collection.block_theme.spec.ts

```diff
@@ -102,53 +104,173 @@ test.describe( 'Product Collection', () => {
+		test.skip( 'Products can be filtered based on stock status (in stock, out of stock, or backorder).', async ( {
+			pageObject,
+		} ) => {
+			await pageObject.setFilterComboboxValue( 'Stock status', [
+				'Out of stock',
+			] );
+
+			await expect( pageObject.products ).toHaveCount( 1 );
+
+			await pageObject.publishAndGoToFrontend();
+
+			await expect( pageObject.products ).toHaveCount( 1 );
+		} );
 	} );
 
-	test( 'Products can be filtered based on selection in handpicked products option', async ( {
-		pageObject,
-	} ) => {
-		await pageObject.addFilter( 'Show Hand-picked Products' );
+	test.describe( 'Toolbar settings', () => {
+		test( 'Toolbar -> Items per page, offset & max page to show', async ( {
+			pageObject,
+		} ) => {
+			await pageObject.setDisplaySettings( {
+				itemsPerPage: 3,
+				offset: 0,
+				maxPageToShow: 2,
+			} );
 
-		await pageObject.setHandpickedProducts( [ 'Album' ] );
-		expect( pageObject.productTitles ).toHaveCount( 1 );
+			expect( await pageObject.products ).toHaveCount( 3 );
 
-		const productNames = [ 'Album', 'Cap' ];
-		await pageObject.setHandpickedProducts( productNames );
-		expect( pageObject.productTitles ).toHaveCount( 2 );
-		expect( pageObject.productTitles ).toHaveText( productNames );
+			await pageObject.setDisplaySettings( {
+				itemsPerPage: 2,
+				offset: 0,
+				maxPageToShow: 2,
+			} );
+			expect( await pageObject.products ).toHaveCount( 2 );
 
-		await pageObject.publishAndGoToFrontend();
-		expect( pageObject.productTitles ).toHaveCount( 2 );
-		expect( pageObject.productTitles ).toHaveText( productNames );
+			await pageObject.publishAndGoToFrontend();
+
+			expect( await pageObject.products ).toHaveCount( 2 );
+
+			const paginationNumbers =
+				pageObject.pagination.locator( '.page-numbers' );
+			await expect( paginationNumbers ).toHaveCount( 2 );
+		} );
 	} );
 
-	test( 'Products can be filtered based on keyword.', async ( {
-		page,
-		admin,
-		editor,
+	test( 'Responsive -> Block correctly adjusts number of columns on smaller screens', async ( {
+		pageObject,
 	} ) => {
-		const pageObject = new ProductCollectionPage( {
-			page,
-			admin,
-			editor,
-		} );
-		await pageObject.createNewPostAndInsertBlock();
-		await pageObject.addFilter( 'Keyword' );
+		await pageObject.publishAndGoToFrontend();
 
-		await pageObject.setKeyword( 'Album' );
-		expect( pageObject.productTitles ).toHaveText( [ 'Album' ] );
+		const firstProduct = pageObject.products.first();
 
-		await pageObject.setKeyword( 'Cap' );
-		expect( pageObject.productTitles ).toHaveText( [ 'Cap' ] );
+		// In the original viewport size, we expect the product width to be less than the parent width
+		// because we will have more than 1 column
+		let productWidth = await firstProduct.boundingBox();
+		let parentWidth = await (
+			await firstProduct.locator( 'xpath=..' )
+		 ).boundingBox();
+		expect( productWidth?.width ).toBeLessThan(
+			parentWidth?.width as number
+		);
```

Comment:
It's a detail, but I'd rename `productWidth` and `parentWidth` into something like `productSize` / `parentSize` or similar. Because it's an object containing size from which you take a width from:

```suggestion
		let productSize = await firstProduct.boundingBox();
		let parentSize = await (
			await firstProduct.locator( 'xpath=..' )
		 ).boundingBox();
		expect( productSize?.width ).toBeLessThan(
			parentSize?.width as number
		);
```

Reply:
Makes sense. Made the changes in [363470d](https://github.com/woocommerce/woocommerce-blocks/pull/10090/commits/363470dcb39eace60bc4970283c04e078d88b86a)